gpt_index/prompts/base.py

```diff
@@ -22,22 +22,39 @@ class Prompt:
     input_variables: List[str]
     prompt_type: PromptType = PromptType.CUSTOM
 
-    def __init__(self, template: str, **prompt_kwargs: Any) -> None:
+    def __init__(self, template: str, langchain_prompt: Optional[LangchainPrompt] = None, **prompt_kwargs: Any) -> None:
```

Comment:
if it's either template or langchain_prompt i'd probably also type template as optional